parsing/lexer.mll

```diff
@@ -385,20 +385,21 @@ rule token = parse
         is_in_string := false;
         lexbuf.lex_start_p <- string_start;
         STRING (get_stored_string(), Some delim) }
-  | "\'" newline "\'"
+  | "'" newline "'"
       { update_loc lexbuf None 1 false 1;
-        CHAR (Lexing.lexeme_char lexbuf 1) }
-  | "\'" [^ '\\' '\'' '\010' '\013'] "\'"
+        (* newline is ('\013'* '\010') *)
+        CHAR '\n' }
+  | "'" [^ '\\' '\'' '\010' '\013'] "'"
       { CHAR(Lexing.lexeme_char lexbuf 1) }
-  | "\'\\" ['\\' '\'' '\"' 'n' 't' 'b' 'r' ' '] "\'"
+  | "'\\" ['\\' '\'' '"' 'n' 't' 'b' 'r' ' '] "'"
       { CHAR(char_for_backslash (Lexing.lexeme_char lexbuf 2)) }
-  | "\'\\" ['0'-'9'] ['0'-'9'] ['0'-'9'] "\'"
+  | "'\\" ['0'-'9'] ['0'-'9'] ['0'-'9'] "'"
```

Comment:
Would "999" actually be valid? If not, perhaps the first range should only cover a more limited number of integers...

Reply:
As in the main compiler lexer, the range check is done in the semantic action for better error reporting.

Reply:
Then why bound octal digits to 0-3 instead of reporting in the semantic action?

Reply:
The difference with octal escape sequences is that by limiting them to `[0-3][0-7][0-7]`, exactly the required range of 0-255 is accepted. For decimal escape sequences you need
`[01][0-9][0-9]|2([0-4][0-9]|5[0-5])`
to do that.